prow/config/config.go

```diff
@@ -2401,20 +2401,21 @@ func validatePodSpec(jobType prowapi.ProwJobType, spec *v1.PodSpec, decorationEn
 	}
 
 	volumeNames := sets.String{}
+	decoratedVolumeNames := decorationConfig.Volumes()
```

Comment:
I don't understand this, why do we not call decorationconfig.Volumes in `func (d *DecorationConfig) Volumes()` instead of forcing the caller to know that both need to be  included?

Reply:
Because the user specifies the name of the volume. Example of
```yaml
    decoration_config:
      oauth_token_secret:
        key: oauth
        name: my-secret
```
we need to allow the user to be able to add in `volumeMounts` the volume named `my-secret` without adding it as a volume since that will happen in runtime.

Reply:
Yeah I understand that, but right now the callsite has to check if decoration is enabled, if so call `decorationConfig.Volumes()`, then call `decorationConfig.VolumeMounts()` (the first being the package, the second the variable) and then assemble them together. Why not call the latter inside the former?

Reply:
Because they are two different things. Do you mean that its better to just refactor [decorate.VolumeMounts()](https://github.com/kubernetes/test-infra/blob/master/prow/pod-utils/decorate/podspec.go#L70-L73) to get a decoration config and extract the volumes from there?

Reply:
A union of those sets should be logical here.

Reply:
the purpose of `decorationConfig.Volumes()` is to return all volume names that will be used by this decoration config or not? So including `decorationconfig.VolumeMounts()` in there seems to make sense.

It maybe be very logical for you working on this code right now, but it requires the caller to know a bunch of implementation details to get this right and that shouldn't be required and is likely to be misused.

Reply:
The `decorate.VolumeMounts()` is not related with any decorationConfig configuration. So even now the user can misuse it. Do you think that its better to drop a different message for the user since he will be responsible for the input?